Gemfile

```diff
@@ -5,7 +5,8 @@ sqlite = ENV['SQLITE_VERSION']
 if sqlite
   gem 'sqlite3', sqlite, platforms: [:ruby]
 else
-  gem 'sqlite3', '~> 1.4', platforms: [:ruby]
+  # Rails 8.0 requires sqlite3 2.x
+  gem 'sqlite3', ENV['RAILS']&.start_with?('~> 8') ? '~> 2.1' : '~> 1.4', platforms: [:ruby]
```

Comment:
I wonder, if `2.1+` works with rails 7 or 6, then we should turn the condition around and only use `~> 1.4` for older versions that don't support 2.x